students/jared_mulholland/lesson_9/mail_room_class.py

```diff
@@ -0,0 +1,243 @@
+"""import required modules"""
+
+import numpy as np
+import os
+import sys
+
+"""
+
+Notes for mail room class testing:
+
+1. Class for individual donor
+    a. contains, first name, last name, donation amount
+    b. method to append to donations
+    c. method to send a thank you
+"""
+
+class Donor(object):
+    """takes full name and creates donor record
+    need a donor and a donation to create a donor record"""
+    def __init__(self, fullname, donation):
+        if len(fullname.split(" ")) == 1:
+            first = "Mr/Ms"
+            last = fullname.split(" ")[0]
+        elif len(fullname.split(" ")) == 2:
+            first, last = fullname.split(" ")
+        else:
+            first = fullname.split(" ")[0]
+            last = fullname.split(" ")[len(fullname.split(" "))-1]
+        self.first = first
+        self.last = last
+        self.fullname = fullname
+        self.donation = [donation]
+
+    @property
+    def add_donation(self):
+        return self.donation
+    
+    @add_donation.setter
+    def add_donation(self, new_donation):
+        self.donation.append(new_donation)
+    
+    @property    
+    def thank_you(self):
+        """sends thank you for latest donation"""
+        newest_donation = self.donation[len(self.donation)-1]        
+        thankyou =  f"""\nDear {self.first}, \n\nThank you for your generous donation of ${newest_donation:,.2f}. Thanks to you, we will finally be able to begin construction on the {self.last} Memorial Children's wing at The Dark Place Hospital\n\nYours Truly,\nGarth Marenghi\n"""
+        print(thankyou)    
+    
+    def __repr__(self):
+        return "Donor Record: {}, {}".format(self.last, self.first)
+
+
+"""
+2. Class for collection of donors
+    a. has list of donors and donotion amounts
+    b. method to add donors
+    c. method to create report
+    d. method to create letters
+"""
+
+class DonorGroup(object):
+    """Donor group initialized with first Donor class
+    Donors are kept in a dictionary created in DonorGroup"""
+    def __init__(self, donor=None, file_path=None):
+        self.donor_list = [donor.fullname]
+        self.donor_dict = {donor.fullname: donor.donation}
+        self.file_path = file_path
+        
+    @property
+    def add_donor(self):
+        return self.donor_list
+    
+    @add_donor.setter
+    def add_donor(self, new_donor):
+        self.donor_list.append(new_donor.fullname)
+        self.donor_dict[new_donor.fullname] = new_donor.donation    
+
+    @property
+    def create_report(self):
+        rows = [(donor, sum(self.donor_dict[donor]),len(self.donor_dict[donor]),np.mean(self.donor_dict[donor])) for donor in self.donor_dict]
```

Comment:
all the "total", "number of" and "average" attributes should be part of your Donor class and accessed as attributes.